nextmv/nextmv/options.py

```diff
@@ -624,10 +627,70 @@ def merge(self, new: "Options") -> "Options":
 
         self.options += new.options
 
-        self._parse()
+        if not skip_parse:
+            self._parse()
 
         return self
 
+    @classmethod
+    def merge_all(cls, *options: "Options", skip_parse: bool = False) -> "Options":
+        """
+        Merges multiple `Options` instances into a single `Options` instance.
+
+        This method cannot be used if any of the options have been parsed already. When
+        options are parsed, values are read from the command-line arguments, environment
+        variables and default values. Merging options after parsing would result in
+        unpredictable behavior.
+
+        Parameters
+        ----------
+        *options : Options
+            The `Options` instances to merge. At least one `Options` instance is required.
+        skip_parse : bool, optional
+            If True, the merged options will not be parsed after merging. This is useful
+            if you want to merge further options after this merge. The default is False.
+
+        Returns
+        -------
+        Options
+            A new `Options` instance containing all options from the provided instances.
+
+        Raises
+        ------
+        ValueError
+            If no `Options` instances are provided to merge.
+        RuntimeError
+            If any of the provided `Options` instances have already been parsed.
+
+        Examples
+        --------
+        >>> opt1 = Options(Option("duration", str, "30s"))
+        >>> opt2 = Options(Option("threads", int, 4))
+        >>> merged = Options.merge_all(opt1, opt2)
+        >>> merged.duration
+        '30s'
+        >>> merged.threads
+        4
+        """
+
+        if not options:
+            raise ValueError("at least one Options instance is required to merge")
+
+        merged_options_list = []
+        for opt in options:
+            if opt.PARSED:
+                raise RuntimeError(
+                    "options have already been parsed, cannot merge. See `Options.parse()` for more information."
```

Comment:
I would add at least the index of the options object that is parsed to the msg for better debugging.